packages/typegpu/src/smol/wgslGenerator.ts

```diff
@@ -428,6 +428,44 @@ ${alternate}`;
     }
   }
 
+  // 'j' stands for for (trust me)
+  if ('j' in statement) {
+    const [init, condition, update, body] = statement.j;
+    const initStatement = init ? generateStatement(ctx, init) : undefined;
+    const initStr = initStatement ? initStatement.slice(0, -1) : '';
+    const conditionExpr = condition
+      ? generateExpression(ctx, condition)
+      : undefined;
+    const conditionStr = conditionExpr ? resolveRes(ctx, conditionExpr) : '';
+    const updateStatement = update ? generateStatement(ctx, update) : undefined;
+    const updateStr = updateStatement ? updateStatement.slice(0, -1) : '';
+    const bodyStr = generateStatement(ctx, body);
```

Comment:
There seems to be a missing call to "indent" and "dedent". You can reference the "if" generation.